### ISSUE TITLE
Update to net.mikehardy:google-analytics-java:2.0.11

### ISSUE THREAD

**Issue by david-allison-1** [ankidroid/Anki-Android#7056]
There are issues due to transitive dependency updates after the analytics update - reverting for now, let's move it to a separate issue.

```
> Task :AnkiDroid:compileDebugJavaWithJavac
C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\main\java\com\ichi2\anki\web\HttpFetcher.java:126: warning: [deprecation] create(MediaType,byte[]) in RequestBody has been deprecated
                requestBuilder.post(RequestBody.create(null, new byte[0]));
                                               ^
error: warnings found and -Werror specified
warnings found and -Werror specified

C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\main\java\com\ichi2\libanki\sync\CountingFileRequestBody.java:64: warning: [deprecation] buffer() in BufferedSink has been deprecated
            while ((read = source.read(sink.buffer(), SEGMENT_SIZE)) != -1) {
                                           ^
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 error
2 warnings

> Task :AnkiDroid:compileDebugJavaWithJavac FAILED
```

_Originally posted by @david-allison-1 in https://github.com/ankidroid/Anki-Android/pull/7055#issuecomment-690363780_

**Comment by mikehardy** [comment]
Yeah, that's unexpected! :thinking:

**Comment by mikehardy** [comment]
I want to link this to #7052

**Comment by mikehardy** [comment]
Would be unblocked by #8361 I believe

**Comment by mikehardy** [comment]
I took another try at this and I was able to fix all the undeclared transitive usages of the java8 objects imported by the java7-compatible flavor of google-analytics-java

I still haven't found a super-easy way to fix this though:

```
com.ichi2.anki.FieldEditLineTest > testSaveRestore[TestingAVD_API-21_TAG-GAPI(AVD) - 5.0.2] FAILED 
        java.lang.NoClassDefFoundError: Failed resolution of: Ljava/util/concurrent/CompletableFuture;
        at com.brsanthu.googleanalytics.internal.GoogleAnalyticsImpl.postAsync(GoogleAnalyticsImpl.java:123)
[InstrumentationResultParser]: Ignoring repeated failed event for com.ichi2.anki.FieldEditLineTest#testSaveRestore. Stack: java.lang.AssertionError: Activity never becomes requested state "[CREATED, STARTED, RESUMED, DESTROYED]" (last lifecycle transition = "STARTED")
``` 

That object is not included in the java8 de-sugaring, so this is effectively on hold until I either find an easy way to package that dependency up *in* the google-analytics-java artifact (not likely, it's work that has little value) or we go to minSdkVersion >= 24.

**Comment by dae** [comment]
@mikehardy still an issue?

**Comment by mikehardy** [comment]
I think desugaring has gotten better, will re-investigate as I don't think we have anything blocking using java8 features now
analytics in general needs a  big push as GA4 happened

**Comment by david-allison** [comment]
@mikehardy We're on API 24 now. Anything pending?

**Comment by mikehardy** [comment]
@david-allison no - the libre wrapper for GA needs a huge update obviously, and it hasn't had one, but I don't believe we should have any issue using the modern-API version 

It should just be a matter of taking the `7`'s off here and uses of this line 

https://github.com/ankidroid/Anki-Android/blob/96ee7c5a21536237eff54aec39786cad294c332a/gradle/libs.versions.toml#L145

**Comment by YashSonbhurra** [comment]
Can I handle that?
